lib/presentation/widgets/loading_indicator/app_loading_indicator.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/material.dart';
+
+class AppLoadingIndicator extends StatelessWidget {
+  /// Determines both height and width of the [AppLoadingIndicator]
+  static const defaultSize = 24.0;
+
+  /// Determines both height and width of the [AppLoadingIndicator.small]
+  static const defaultSmallSize = 16.0;
+
+  const AppLoadingIndicator({
+    super.key,
+    this.size = defaultSize,
+  });
+
+  /// Determines both height and width of the loader.
+  final double size;
+
+  factory AppLoadingIndicator.small() {
+    return const AppLoadingIndicator(
+      size: defaultSmallSize,
```

Comment:
Consider making the thickness smaller for smaller material indicators. Having tiny but thick indicators might be weird